### PR TITLE
Upload per target coverage report.

### PR DESCRIPTION
**Pull request by oliverchang**
For #7015.

### CODE REVIEW

**Comment by oliverchang** [comment]
Example report: https://storage.googleapis.com/oss-fuzz-coverage/abseil-cpp/reports-by-target/20220316/string_utilities_fuzzer/linux/report.html

File: infra/build/functions/build_and_run_coverage.py [hunk 1]
```diff
@@ -173,6 +174,19 @@ def get_build_steps(  # pylint: disable=too-many-locals, too-many-arguments
...
 
+  build_steps.append(build_lib.gsutil_rm_rf_step(upload_report_by_target_url))
+  build_steps.append({
+      'name':
+          'gcr.io/cloud-builders/gsutil',
+      'args': [
+          '-m',
+          'cp',
+          '-r',
+          os.path.join(build.out, 'report_target'),
```
Comment (jonathanmetzman) [inline infra/build/functions/build_and_run_coverage.py:185]: nit: `target_report` sounds better to me.

**Review by jonathanmetzman — APPROVED** [review]
lgtm

**Review by Navidem — APPROVED** [review]

File: infra/build/functions/build_and_run_coverage.py (hunk 1 above)
Comment (oliverchang) [reply infra/build/functions/build_and_run_coverage.py:185]: This isn't really visible to users, and is a bit annoying to change at this point (requires a base image change first). Can change later.

**Comment by fmeum** [comment]
@oliverchang @jonathanmetzman Java coverage builds are failing: https://oss-fuzz-build-logs.storage.googleapis.com/log-8fca09a3-1981-450d-a942-4af391e812ac.txt
Could this be related to this PR?

**Comment by DavidKorczynski** [comment]
go builds are failing as well

**Comment by jonathanmetzman** [comment]
See https://github.com/google/oss-fuzz/pull/7389

**Comment by jonathanmetzman** [comment]
Deploying this too

**Pull request merged**